### PR TITLE
pushing upstream openbsd support from package.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by MarioLiebisch** [comment]
Is it intentional to use the Android joystick implementation? If so, it might be a good idea to rename or move it.

**Comment by devnexen** [comment]
Yes intentional :-) but it concerns only few files among Android implementations in another hand ... what do you think ?

File: src/SFML/Window/CMakeLists.txt [hunk 1]
```diff
@@ -116,6 +116,12 @@ elseif(SFML_OS_LINUX OR SFML_OS_FREEBSD)
             ${SRCROOT}/FreeBSD/JoystickImpl.cpp
             ${SRCROOT}/FreeBSD/JoystickImpl.hpp
         )
+    elseif(SFML_OS_OPENBSD)
+        set(PLATFORM_SRC
+            ${PLATFORM_SRC}
+            ${SRCROOT}/Android/JoystickImpl.cpp
```
Comment (mantognini) [inline src/SFML/Window/CMakeLists.txt]: Since the android implementation is a dummy one, I'd rather see those files being copied to a new directory.

**Comment by mantognini** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Window/CMakeLists.txt (hunk 1 above)
Comment (devnexen) [reply src/SFML/Window/CMakeLists.txt]: Ok will do later in the day fair enough :-)

**Comment by mantognini** [comment]
SFMLBuildMaster: Build this please.

**Comment by eXpl0it3r** [comment]
@devnexen We've gotten quite some changes with our CMake setup. Could you resolve the merge conflicts?

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Comment by binary1248** [comment]
The rebase doesn't look clean. Stuff like `INSTALL_MISC_DIR` that was removed in ee08e18726788e4855218fa27319b2313b185d0f are added back in here.

**Comment by binary1248** [comment]
Browsing through the SDL source code, I noticed that [SDL shares the same joystick code among all BSDs](http://hg.libsdl.org/SDL/file/fbfacc66c65c/src/joystick/bsd/SDL_sysjoystick.c). SFML's current FreeBSD Joystick implementation makes use of usbhid instead of the joy driver. usbhid should be available on OpenBSD as well no?

**Comment by devnexen** [comment]
This code is not compiled for OpenBSD (e.g. SDL_JOYSTICK_USBHID is set to zero in our case)

**Comment by binary1248** [comment]
I'm not an expert in OpenBSD by any means, but does this come down to whether usbhid is installed by the user? Or is it not available at all as a package on OpenBSD? [There seem to be OpenBSD man pages for usbhid with the remark that usbhid has been available since OpenBSD 3.0](https://man.openbsd.org/usbhid.3).

**Comment by devnexen** [comment]
usbhid is available but no joystick support. The SDL code you mentioned earlier needs joystick.h header which we di not have.

**Comment by devnexen** [comment]
To come back to SFML until now we used empty android implementation  as you can see here https://cvsweb.openbsd.org/cgi-bin/cvsweb/ports/multimedia/sfml/patches/patch-src_SFML_Window_JoystickImpl_hpp?rev=1.2&content-type=text/x-cvsweb-markup

**Comment by binary1248** [comment]
OK I guess. 👍

File: cmake/Config.cmake [hunk 2]
```diff
@@ -117,3 +121,12 @@ else()
     message(FATAL_ERROR "Unsupported compiler")
     return()
 endif()
+
+# define the install directory for miscellaneous files
```
Comment (binary1248) [inline cmake/Config.cmake]: This block is no longer used in the current master revision.

File: include/SFML/Config.hpp [hunk 3]
```diff
@@ -85,6 +85,10 @@
 
         // FreeBSD
         #define SFML_SYSTEM_FREEBSD
+    #elif defined(__OpenBSD__)
```
Comment (binary1248) [inline include/SFML/Config.hpp]: Empty line missing before this one.

File: src/SFML/Window/CMakeLists.txt [hunk 4]
```diff
@@ -259,7 +265,7 @@ if(SFML_OS_LINUX)
     target_link_libraries(sfml-window PRIVATE UDev)
 elseif(SFML_OS_WINDOWS)
     target_link_libraries(sfml-window PRIVATE winmm gdi32)
-elseif(SFML_OS_FREEBSD)
+elseif(SFML_OS_FREEBSD OR SFML_OS_OPENBSD)
```
Comment (binary1248) [inline src/SFML/Window/CMakeLists.txt]: If we can't interface with joysticks via usbhid do we still need to link to it on OpenBSD?

File: src/SFML/Window/JoystickImpl.hpp [hunk 5]
```diff
@@ -98,7 +98,7 @@ struct JoystickState
 
     #include <SFML/Window/iOS/JoystickImpl.hpp>
 
-#elif defined(SFML_SYSTEM_ANDROID)
+#elif defined(SFML_SYSTEM_ANDROID) || defined(SFML_SYSTEM_OPENBSD)
```
Comment (binary1248) [inline src/SFML/Window/JoystickImpl.hpp]: This should be split out and include the OpenBSD implementation instead.

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Window/OpenBSD/JoystickImpl.hpp [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+#ifndef SFML_JOYSTICKIMPLOPENBSD_HPP
+#define SFML_JOYSTICKIMPLOPENBSD_HPP
+
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+/// \brief Android implementation of joysticks
```
Comment (Foaly) [inline src/SFML/Window/OpenBSD/JoystickImpl.hpp]: Copy pasta :)

**Comment by eXpl0it3r** [comment]
Merged in 9da895da8baae8d82bf2c870a50f987a03fd8f18

**Pull request closed without merging**